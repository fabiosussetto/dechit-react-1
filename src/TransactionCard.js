import React from 'react'
import ExpandCollapse from './ExpandCollapse';

const TransactionCard = (props) => {
    const { transaction, onRemoveTransaction, onIncrementAmount, expanded, onToggleExpand } = props

    return (
        <div className="card p-2 mb-2">
            <div className="d-flex">
                <div>
                    
                    <p className="card-text">
                        Amount: {transaction.amount}
                    </p>
                </div>
                <div className="ml-auto">
                    <button className="btn btn-success" onClick={onIncrementAmount}>Add 10</button>
                    <button className="btn btn-secondary" onClick={onToggleExpand}>
                      {expanded ? 'Collapse' : 'Expand'}
                    </button>
                    <button className="btn btn-danger" onClick={onRemoveTransaction}>Remove</button>
                </div>
            </div>
            {expanded && <ExpandCollapse transaction={transaction} />}
        </div>
    )
}

export default TransactionCard