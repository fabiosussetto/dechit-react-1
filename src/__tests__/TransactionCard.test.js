import React from 'react';
import TransactionCard from '../TransactionCard';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const transaction = {
      id: 1,
      amount: 100,
      category: 'books',
      description: 'test'
  }

  const tree = renderer
    .create(<TransactionCard transaction={transaction} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});