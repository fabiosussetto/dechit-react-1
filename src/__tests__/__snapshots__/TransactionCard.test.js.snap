// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card p-2 mb-2"
>
  <div
    className="d-flex"
  >
    <div>
      <p
        className="card-text"
      >
        Amount: 
        100
      </p>
    </div>
    <div
      className="ml-auto"
    >
      <button
        className="btn btn-success"
      >
        Add 10
      </button>
      <button
        className="btn btn-secondary"
      >
        Expand
      </button>
      <button
        className="btn btn-danger"
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
